external/llvm-project/mlir/include/mlir/Dialect/Tosa/Transforms/Passes.h

```diff
@@ -31,9 +31,9 @@ std::unique_ptr<Pass> createTosaInferShapesPass();
 std::unique_ptr<Pass> createTosaMakeBroadcastablePass();
 std::unique_ptr<Pass> createTosaTestQuantUtilAPIPass();
 std::unique_ptr<Pass> createTosaPartitionPass();
-std::unique_ptr<Pass> createPostPartitionCollapsePass();
+class PartitionConfig;
```

Comment:
Remove the obsolete declarations for passes we no longer have, and add a second variant of createTosaPartitionPass() to allow specifying the partitioning configuration at the program level. Typical command-line usage will call the no-arguments version and interpret the command-line options; calling within a program can supply the configuration as an argument.  PartitionConfig is forward-declared because that's the abstract interface for users that write their own.